src/decoders/sndfile.cpp

```diff
@@ -128,10 +128,20 @@ std::pair<uint64_t, uint64_t> SndFileDecoder::getLoopPoints() const noexcept { r
 ALuint SndFileDecoder::read(ALvoid *ptr, ALuint count) noexcept
 {
     sf_count_t got = 0;
-    if(mSampleType == SampleType::Int16)
-        got = sf_readf_short(mSndFile.get(), static_cast<short*>(ptr), count);
-    else if(mSampleType == SampleType::Float32)
-        got = sf_readf_float(mSndFile.get(), static_cast<float*>(ptr), count);
+    switch (mSampleType)
+    {
+        case SampleType::UInt8:
+            got = sf_read_raw(mSndFile.get(), static_cast<ALubyte *>(ptr),
+                FramesToBytes(count, mChannelConfig, mSampleType));
+            got = BytesToFrames(got, mChannelConfig, mSampleType);
+            break;
+        case SampleType::Int16:
+            got = sf_readf_short(mSndFile.get(), static_cast<short *>(ptr), count);
+            break;
+        case SampleType::Float32:
+            got = sf_readf_float(mSndFile.get(), static_cast<float *>(ptr), count);
+            break;
+    }
```

Comment:
This will generate a warning with some compilers, since the `SampleType::Mulaw` case isn't handled for the `SampleType` variable. It would probably be best to add the extra case for it, along with a comment explaining muLaw (aka μ-law, sometimes called u-law) samples aren't read from libsndfile, rather than a catch-all `default` case.

Alternatively, you could add support for muLaw as well. If the file format is muLaw and if muLaw samples are supported (see the `Float32` format check), set that sample type and handle it with the raw function like uint8.

Reply:
Understood. Done. Thankfully libsndfile has full support for muLaw.

I am surprised with the quality of muLaw, a lot less noisier than plain 8-bit audio. Really impressive.